components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java

```diff
@@ -196,7 +205,107 @@ public Response logout(@Context Request request) {
         errorDTO.setCode(ExceptionCodes.INVALID_AUTHORIZATION_HEADER.getErrorCode());
         errorDTO.setMessage(ExceptionCodes.INVALID_AUTHORIZATION_HEADER.getErrorMessage());
         return Response.status(Response.Status.UNAUTHORIZED).entity(errorDTO).build();
+    }
 
+    /**
+     * This method provides the DCR application information to the SSO-IS login.
+     *
+     * @param request Request to call the /login api
+     * @return Response - Response object with OAuth data
+     */
+    @GET
+    @Path("/login/{appName}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response redirect(@Context Request request) {
+        String appContext = AuthUtil.getAppContext(request);
+        String appName = appContext.substring(1);
+        try {
+            KeyManager keyManager = APIManagerFactory.getInstance().getKeyManager();
+            AuthenticatorService authenticatorService = new AuthenticatorService(keyManager);
+            JsonObject oAuthData = authenticatorService.getDCRApplicationDetails(appName);
+            if (oAuthData.size() == 0) {
```

Comment:
Yes, this value can be zero, when OAuthApplicationInfo object is null at getAuthenticationConfigurations().